<p>
<img alt="CC" height="70px" src="https://creativecommons.org/images/deed/cc_blue_x2.png" align="left" hspace="0px" vspace="0px">
<img alt="Attribution" height="70px" src="https://creativecommons.org/images/deed/attribution_icon_blue_x2.png" align="left" hspace="0px" vspace="0px">
<img alt="NC" height="70px" src="https://creativecommons.org/images/deed/nc_blue_x2.png" align="left" hspace="0px" vspace="0px">
<img alt="SA" height="70px" src="https://creativecommons.org/images/deed/sa_blue_x2.png" align="left" hspace="0px" vspace="0px">
</p>

<div align="right">
<h1> <b> Elaborado por Edgar Castilla. </b> </h1>

**Documentos de ayuda**:
  
*   Python Website. (2021). Python 3.9.1 documentation. Retrieved February 03, 2021, from https://docs.python.org/
*   W3schools.com. (2021). Python tutorial. Retrieved February 04, 2021, from https://www.w3schools.com/python/default.asp
*   realpython.com. (2021). Real Python Tutorials. Retrieved February 04, 2021, from https://realpython.com/
* https://plotly.com/python/

**Importamos las librerías**

In [1]:
import numpy as np
import pandas as pd


**Datos de créditos de consumo otorgado en Medellín obtenidos de la página de datos abiertos de la Alcaldía de Medellín.**

 http://medata.gov.co<br>
 https://acortar.link/zGU2gK

**Lectura del archivo con extensión o formato csv separado por ;**

**df.head(5): Muestra las primeros 5 filas del archivo créditos de consumo otorgados**

In [2]:
df = pd.read_csv('creditos_de_consumo_otorgados.csv',sep=';')
df.head(5)

edad       sexo cabeza_de_hogar caracterizacion_ciudadano  estrato  \
0    45  masculino              si             independiente      1.0   
1    45   femenino              no                      otro      3.0   
2    28   femenino              no                      otro      3.0   
3    30   femenino              si               emprendedor      3.0   
4    36   femenino              no             independiente      2.0   

        barrio_vereda         comuna               zona  \
0             no dato  13-san javier  centro occidental   
1  belen las mercedes       16-belén      suroccidental   
2      belen la palma       16-belén      suroccidental   
3   manrique oriental    03-manrique        nororiental   
4     san javier no.1  13-san javier  centro occidental   

   fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
0                      29                      12                    2017   
1                      11                      12                    2017   
2                      11                      12                    2017   
3                      11                      12                    2017   
4                      11                      12                    2017   

   monto_prestado  
0        200000.0  
1        200000.0  
2        200000.0  
3        200000.0  
4        200000.0

**df.tail(): Muestra las últimas 5 filas del archivo créditos de consumo otorgados**

In [ ]:
df.tail()

edad      sexo        cabeza_de_hogar caracterizacion_ciudadano  \
12147    57  femenino  aporta economicamente                       NaN   
12148    53  femenino  aporta economicamente                       NaN   
12149    70  femenino  aporta economicamente                       NaN   
12150    70  femenino  aporta economicamente                       NaN   
12151    70  femenino  aporta economicamente                       NaN   

       estrato    barrio_vereda              comuna           zona  \
12147      2.0  doce de octubre  06-doce de octubre  noroccidental   
12148      3.0  doce de octubre  06-doce de octubre  noroccidental   
12149      2.0  doce de octubre  06-doce de octubre  noroccidental   
12150      2.0  doce de octubre  06-doce de octubre  noroccidental   
12151      2.0  doce de octubre  06-doce de octubre  noroccidental   

       fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
12147                      29                       6                    2019   
12148                      29                       6                    2019   
12149                      29                       6                    2019   
12150                      29                       6                    2019   
12151                      29                       6                    2019   

       monto_prestado  
12147        300000.0  
12148        300000.0  
12149        300000.0  
12150        300000.0  
12151        300000.0

In [ ]:
print(df.info())
print("\nConteo de valores únicos:\n")
print(df.nunique())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12152 entries, 0 to 12151
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   edad                       12152 non-null  int64  
 1   sexo                       12152 non-null  object 
 2   cabeza_de_hogar            12034 non-null  object 
 3   caracterizacion_ciudadano  9133 non-null   object 
 4   estrato                    10599 non-null  float64
 5   barrio_vereda              12152 non-null  object 
 6   comuna                     10717 non-null  object 
 7   zona                       10715 non-null  object 
 8   fecha_de_beneficio_dia     12152 non-null  int64  
 9   fecha_de_beneficio_mes     12152 non-null  int64  
 10  fecha_de_beneficio_año     12152 non-null  int64  
 11  monto_prestado             12152 non-null  float64
dtypes: float64(2), int64(4), object(6)
memory usage: 1.1+ MB
None

Conteo de valores únicos:

edad        

**Nombre de las Columnas.**

In [ ]:
df.columns

Index(['edad', 'sexo', 'cabeza_de_hogar', 'caracterizacion_ciudadano',
       'estrato', 'barrio_vereda', 'comuna', 'zona', 'fecha_de_beneficio_dia',
       'fecha_de_beneficio_mes', 'fecha_de_beneficio_año', 'monto_prestado'],
      dtype='object')

**Valores nulos en el archivo.**

In [ ]:
print("Hay algun valor nulo en los datos? :",df.isnull().any().any())
print(df.isnull().sum())

Hay algun valor nulo en los datos? : True
edad                            0
sexo                            0
cabeza_de_hogar               118
caracterizacion_ciudadano    3019
estrato                      1553
barrio_vereda                   0
comuna                       1435
zona                         1437
fecha_de_beneficio_dia          0
fecha_de_beneficio_mes          0
fecha_de_beneficio_año          0
monto_prestado                  0
dtype: int64


**Dimensión de los datos (filas y columnas)**

In [ ]:
df.shape

(12152, 12)

In [ ]:
df['sexo'].value_counts()

femenino     8069
masculino    4083
Name: sexo, dtype: int64

In [ ]:
df['estrato'].value_counts()

3.0    4482
2.0    4477
1.0    1238
4.0     309
0.0      51
5.0      36
6.0       6
Name: estrato, dtype: int64

In [ ]:
df['zona'].value_counts()

nororiental          3063
centro oriental      2750
noroccidental        2473
centro occidental     992
suroccidental         973
corregimiento         439
suroriental            25
Name: zona, dtype: int64

**Ordenar los valores de la columna zona de forma porcentual de mayor a menor.**

In [ ]:
(df['zona'].value_counts() / len(df)).sort_values(ascending=False)

nororiental          0.252057
centro oriental      0.226300
noroccidental        0.203506
centro occidental    0.081633
suroccidental        0.080069
corregimiento        0.036126
suroriental          0.002057
Name: zona, dtype: float64

**Ordenar los valores de la columna zona de forma porcentual de menor a mayor.**

In [ ]:
(df['zona'].value_counts() / len(df)).sort_values(ascending=True)

suroriental          0.002057
corregimiento        0.036126
suroccidental        0.080069
centro occidental    0.081633
noroccidental        0.203506
centro oriental      0.226300
nororiental          0.252057
Name: zona, dtype: float64

In [ ]:
df['comuna'].value_counts()

10-la candelaria           1361
07-robledo                 1091
04-aranjuez                 974
03-manrique                 930
05-castilla                 925
08-villa hermosa            722
02-santa cruz               715
13-san javier               709
09-buenos aires             667
16-belén                    663
06-doce de octubre          457
01-popular                  444
15-guayabal                 310
60-san cristobal            194
12-la américa               164
80-san antonio de prado     163
11-laureles                 119
90-santa elena               51
70-altavista                 31
14-poblado                   25
22                            1
24                            1
Name: comuna, dtype: int64

In [ ]:
df['cabeza_de_hogar'].value_counts()

no                       6413
si                       3987
aporta economicamente    1466
0                         167
solidaria                   1
Name: cabeza_de_hogar, dtype: int64

In [ ]:
df['caracterizacion_ciudadano'].value_counts()

otro                  3525
independiente         2885
emprendedor           2646
empresario              62
u.t aseo colombia        4
ama de casa              2
indufiestas              2
empleado                 1
coosiendo                1
cootranscataluña         1
expertos seguridad       1
noel                     1
empremdedor              1
no                       1
Name: caracterizacion_ciudadano, dtype: int64

**Estadísticas básicas.**

In [3]:
df.describe()

edad       estrato  fecha_de_beneficio_dia  \
count  12152.000000  10599.000000            12152.000000   
mean      44.789911      2.367204               16.341919   
std       15.763499      0.761825                8.090389   
min       18.000000      0.000000                1.000000   
25%       31.000000      2.000000               10.000000   
50%       45.000000      2.000000               16.000000   
75%       57.000000      3.000000               23.000000   
max       84.000000      6.000000               31.000000   

       fecha_de_beneficio_mes  fecha_de_beneficio_año  monto_prestado  
count            12152.000000            12152.000000    12152.000000  
mean                 4.811224             2017.962558   223368.018186  
std                  2.857677                0.526631    51243.067559  
min                  1.000000             2017.000000    50000.000000  
25%                  3.000000             2018.000000   200000.000000  
50%                  5.000000             2018.000000   200000.000000  
75%                  6.000000             2018.000000   200012.000000  
max                 12.000000             2019.000000   450000.000000

**Estadísticas de una columna numérica**

In [4]:
df['edad'].describe()

count    12152.000000
mean        44.789911
std         15.763499
min         18.000000
25%         31.000000
50%         45.000000
75%         57.000000
max         84.000000
Name: edad, dtype: float64

**Datos categóricos**

**Ordinales y nominales**

In [ ]:
df.dtypes

edad                           int64
sexo                          object
cabeza_de_hogar               object
caracterizacion_ciudadano     object
estrato                      float64
barrio_vereda                 object
comuna                        object
zona                          object
fecha_de_beneficio_dia         int64
fecha_de_beneficio_mes         int64
fecha_de_beneficio_año         int64
monto_prestado               float64
dtype: object

**Categóricos:object y booleano(buleanos o logicos,verdaero o falso,True,False)**

**numéricos: int64, significa valores enteros o discretos,representa un entero de 64 bits** <br>
**numéricos: float64(continuos), significa que toman valores reales de 64 bits**<br>
**En 64 bits, el rango es de -9.223.372.036.854.775.808 hasta 9.223.372.036.854.775.807**

**Monto prestado es una variable numérica continua toma valores como :200000,300000,200012,etc.**

In [ ]:
df['monto_prestado'].value_counts()

200000.0    8117
300000.0    2115
200012.0    1549
450000.0     184
400000.0     136
250000.0      16
100000.0      10
150000.0       8
50000.0        7
350000.0       6
199975.0       2
199606.0       1
200013.0       1
Name: monto_prestado, dtype: int64

**Edad es una variable numérica entera o discreta toma valores:70,55,53,45,28, etc.**

In [ ]:
df['edad'].value_counts()

70    514
55    290
53    286
45    281
28    276
     ... 
82     26
81     25
74     22
75     20
77     20
Name: edad, Length: 67, dtype: int64



*   **Categórico Ordinal: valores representan categorías con alguna clasificación intrínseca ,tiene caracteristicas de orden, ejemplo:malo, bueno, excelente**
*  **Categórico nominal: representan categorías sin clasificación intrínseca,ejemplo:las marcas de zapatos,genero, raza, etnia**

*   **La edad (en años) puede ser tratada :**
*   **nivel ordinal: (niño/adolescente/joven)**

*  **nominal: (adulto/no adulto)**









**Caracterización del ciudadano es un dato categórico nominal toma valores como :otro,independiente,emprendedor,empresario,etc.**

In [ ]:
df['caracterizacion_ciudadano'].value_counts()

otro                  3525
independiente         2885
emprendedor           2646
empresario              62
u.t aseo colombia        4
ama de casa              2
indufiestas              2
empleado                 1
coosiendo                1
cootranscataluña         1
expertos seguridad       1
noel                     1
empremdedor              1
no                       1
Name: caracterizacion_ciudadano, dtype: int64

**Cabeza de hogar es un dato categórico nominal toma los valores:<br>no,si,aporta economicamente,0,solidaria**

In [ ]:
df['cabeza_de_hogar'].value_counts()

no                       6413
si                       3987
aporta economicamente    1466
0                         167
solidaria                   1
Name: cabeza_de_hogar, dtype: int64

In [ ]:
df['cabeza_de_hogar'].unique()

array(['si', 'no', '0', 'solidaria', nan, 'aporta economicamente'],
      dtype=object)

**Remplazar el nan por no tiene valor**

In [ ]:
df['cabeza_de_hogar']=df['cabeza_de_hogar'].fillna('no_tiene_valor')

In [ ]:
df['cabeza_de_hogar'].unique()

array(['si', 'no', '0', 'solidaria', 'no_tiene_valor',
       'aporta economicamente'], dtype=object)

**sexo es una variable categórica nominal toma los valores:femenino y masculino**

In [ ]:
df['sexo'].value_counts()

femenino     8069
masculino    4083
Name: sexo, dtype: int64

**Columnas numéricas**

In [ ]:
columnas_numericas=['edad','estrato','fecha_de_beneficio_dia','fecha_de_beneficio_mes','fecha_de_beneficio_año','monto_prestado']

In [ ]:
datos_numericos=df[columnas_numericas]

In [ ]:
datos_numericos

edad  estrato  fecha_de_beneficio_dia  fecha_de_beneficio_mes  \
0        45      1.0                      29                      12   
1        45      3.0                      11                      12   
2        28      3.0                      11                      12   
3        30      3.0                      11                      12   
4        36      2.0                      11                      12   
...     ...      ...                     ...                     ...   
12147    57      2.0                      29                       6   
12148    53      3.0                      29                       6   
12149    70      2.0                      29                       6   
12150    70      2.0                      29                       6   
12151    70      2.0                      29                       6   

       fecha_de_beneficio_año  monto_prestado  
0                        2017        200000.0  
1                        2017        200000.0  
2                        2017        200000.0  
3                        2017        200000.0  
4                        2017        200000.0  
...                       ...             ...  
12147                    2019        300000.0  
12148                    2019        300000.0  
12149                    2019        300000.0  
12150                    2019        300000.0  
12151                    2019        300000.0  

[12152 rows x 6 columns]

In [ ]:
datos_numericos.dtypes

edad                        int64
estrato                   float64
fecha_de_beneficio_dia      int64
fecha_de_beneficio_mes      int64
fecha_de_beneficio_año      int64
monto_prestado            float64
dtype: object

In [ ]:
df.head(5)

edad       sexo cabeza_de_hogar caracterizacion_ciudadano  estrato  \
0    45  masculino              si             independiente      1.0   
1    45   femenino              no                      otro      3.0   
2    28   femenino              no                      otro      3.0   
3    30   femenino              si               emprendedor      3.0   
4    36   femenino              no             independiente      2.0   

        barrio_vereda         comuna               zona  \
0             no dato  13-san javier  centro occidental   
1  belen las mercedes       16-belén      suroccidental   
2      belen la palma       16-belén      suroccidental   
3   manrique oriental    03-manrique        nororiental   
4     san javier no.1  13-san javier  centro occidental   

   fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
0                      29                      12                    2017   
1                      11                      12                    2017   
2                      11                      12                    2017   
3                      11                      12                    2017   
4                      11                      12                    2017   

   monto_prestado  
0        200000.0  
1        200000.0  
2        200000.0  
3        200000.0  
4        200000.0

In [ ]:
df

edad sexo        cabeza_de_hogar caracterizacion_ciudadano  estrato  \
0        45    M                     si             independiente      1.0   
1        45    F                     no                      otro      3.0   
2        28    F                     no                      otro      3.0   
3        30    F                     si               emprendedor      3.0   
4        36    F                     no             independiente      2.0   
...     ...  ...                    ...                       ...      ...   
12147    57    F  aporta economicamente                       NaN      2.0   
12148    53    F  aporta economicamente                       NaN      3.0   
12149    70    F  aporta economicamente                       NaN      2.0   
12150    70    F  aporta economicamente                       NaN      2.0   
12151    70    F  aporta economicamente                       NaN      2.0   

            barrio_vereda              comuna               zona  \
0                 no dato       13-san javier  centro occidental   
1      belen las mercedes            16-belén      suroccidental   
2          belen la palma            16-belén      suroccidental   
3       manrique oriental         03-manrique        nororiental   
4         san javier no.1       13-san javier  centro occidental   
...                   ...                 ...                ...   
12147     doce de octubre  06-doce de octubre      noroccidental   
12148     doce de octubre  06-doce de octubre      noroccidental   
12149     doce de octubre  06-doce de octubre      noroccidental   
12150     doce de octubre  06-doce de octubre      noroccidental   
12151     doce de octubre  06-doce de octubre      noroccidental   

       fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
0                          29                      12                    2017   
1                          11                      12                    2017   
2                          11                      12                    2017   
3                          11                      12                    2017   
4                          11                      12                    2017   
...                       ...                     ...                     ...   
12147                      29                       6                    2019   
12148                      29                       6                    2019   
12149                      29                       6                    2019   
12150                      29                       6                    2019   
12151                      29                       6                    2019   

       monto_prestado  
0            200000.0  
1            200000.0  
2            200000.0  
3            200000.0  
4            200000.0  
...               ...  
12147        300000.0  
12148        300000.0  
12149        300000.0  
12150        300000.0  
12151        300000.0  

[12152 rows x 12 columns]

**Remplazar el valor de una columna**

In [ ]:
df['sexo']=df['sexo'].replace('masculino','M')
df['sexo']=df['sexo'].replace('femenino','F')

In [ ]:
df.head(5)

edad sexo cabeza_de_hogar caracterizacion_ciudadano  estrato  \
0    45    M              si             independiente      1.0   
1    45    F              no                      otro      3.0   
2    28    F              no                      otro      3.0   
3    30    F              si               emprendedor      3.0   
4    36    F              no             independiente      2.0   

        barrio_vereda         comuna               zona  \
0             no dato  13-san javier  centro occidental   
1  belen las mercedes       16-belén      suroccidental   
2      belen la palma       16-belén      suroccidental   
3   manrique oriental    03-manrique        nororiental   
4     san javier no.1  13-san javier  centro occidental   

   fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
0                      29                      12                    2017   
1                      11                      12                    2017   
2                      11                      12                    2017   
3                      11                      12                    2017   
4                      11                      12                    2017   

   monto_prestado  
0        200000.0  
1        200000.0  
2        200000.0  
3        200000.0  
4        200000.0

**Remplazar varios valores de un columna**

In [5]:
df['caracterizacion_ciudadano']=df['caracterizacion_ciudadano'].replace(['independiente','emprendedor'],['I','E'])

In [6]:
df.head(5)

edad       sexo cabeza_de_hogar caracterizacion_ciudadano  estrato  \
0    45  masculino              si                         I      1.0   
1    45   femenino              no                      otro      3.0   
2    28   femenino              no                      otro      3.0   
3    30   femenino              si                         E      3.0   
4    36   femenino              no                         I      2.0   

        barrio_vereda         comuna               zona  \
0             no dato  13-san javier  centro occidental   
1  belen las mercedes       16-belén      suroccidental   
2      belen la palma       16-belén      suroccidental   
3   manrique oriental    03-manrique        nororiental   
4     san javier no.1  13-san javier  centro occidental   

   fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
0                      29                      12                    2017   
1                      11                      12                    2017   
2                      11                      12                    2017   
3                      11                      12                    2017   
4                      11                      12                    2017   

   monto_prestado  
0        200000.0  
1        200000.0  
2        200000.0  
3        200000.0  
4        200000.0

**Renombrar unas columnas**

In [ ]:
datos=datos_numericos.rename(columns={'fecha_de_beneficio_dia':'fecha de beneficio dia','fecha_de_beneficio_mes':'fecha de beneficio mes','fecha_de_beneficio_año':'fecha de beneficio año'})

In [ ]:
datos.dtypes

edad                        int64
estrato                   float64
fecha de beneficio dia      int64
fecha de beneficio mes      int64
fecha de beneficio año      int64
monto_prestado            float64
dtype: object

**Muestreo aleatorio simple.**

**Importar la librería**

In [ ]:
import random

In [ ]:
muestreo10=df.sample(n=10)

**Toma 10 muestras aleatorias de todo la  población, en este caso 12152**

In [ ]:
muestreo10

edad       sexo        cabeza_de_hogar caracterizacion_ciudadano  \
9263     36  masculino                     no                       NaN   
1335     27  masculino                     no               emprendedor   
10100    32   femenino                     no                       NaN   
6998     69  masculino                     si             independiente   
10727    79  masculino  aporta economicamente                       NaN   
5081     37   femenino                     no               emprendedor   
2593     52   femenino                     no               emprendedor   
904      35   femenino                     si             independiente   
6701     48   femenino                     si                      otro   
2866     24  masculino                     si                      otro   

       estrato          barrio_vereda            comuna             zona  \
9263       NaN                no dato               NaN              NaN   
1335       3.0      campo valdes no.1       04-aranjuez      nororiental   
10100      NaN                no dato               NaN              NaN   
6998       4.0           prado centro  10-la candelaria  centro oriental   
10727      3.0     comuna 04 aranjuez       04-aranjuez      nororiental   
5081       3.0                 gerona   09-buenos aires  centro oriental   
2593       3.0            el salvador   09-buenos aires  centro oriental   
904        3.0  francisco antonio zea  10-la candelaria  centro oriental   
6701       3.0             cristo rey       15-guayabal    suroccidental   
2866       2.0              villatina     02-santa cruz      nororiental   

       fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
9263                       21                      11                    2017   
1335                       16                       1                    2018   
10100                      12                      10                    2017   
6998                       24                       5                    2018   
10727                      26                       2                    2019   
5081                       23                       4                    2018   
2593                       15                       2                    2018   
904                         4                       1                    2018   
6701                       22                       5                    2018   
2866                       21                       2                    2018   

       monto_prestado  
9263         200012.0  
1335         200000.0  
10100        200012.0  
6998         300000.0  
10727        300000.0  
5081         200000.0  
2593         200000.0  
904          200000.0  
6701         200000.0  
2866         200000.0

**Se repite la toma de muestra aleatoria con el mismo n=10, al ser aleatoria selecciona otros datos de la población.**

In [ ]:
muestreo10_nuevamente=df.sample(n=10)

In [ ]:
muestreo10_nuevamente

edad       sexo cabeza_de_hogar caracterizacion_ciudadano  estrato  \
6882    64  masculino              si                      otro      1.0   
9097    48   femenino              no                      otro      2.0   
8580    40   femenino              si             independiente      2.0   
8848    53   femenino              no             independiente      2.0   
1956    32  masculino              si               emprendedor      3.0   
1184    19   femenino              no               emprendedor      2.0   
745     51   femenino              no                      otro      4.0   
6618    59   femenino              si             independiente      1.0   
6473    53   femenino              no                      otro      2.0   
6486    56   femenino              no                      otro      2.0   

                  barrio_vereda              comuna             zona  \
6882  santo domingo savio no. 1          01-popular      nororiental   
9097  santo domingo savio no. 1          01-popular      nororiental   
8580                   manrique    10-la candelaria  centro oriental   
8848       doce de octubre no.2  06-doce de octubre    noroccidental   
1956                   castilla         05-castilla    noroccidental   
1184                 santa cruz       02-santa cruz      nororiental   
745                      boston    10-la candelaria  centro oriental   
6618                    picacho  06-doce de octubre    noroccidental   
6473    playon de los comuneros       02-santa cruz      nororiental   
6486                 villa niza       02-santa cruz      nororiental   

      fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
6882                      22                       5                    2018   
9097                      30                       8                    2018   
8580                      15                       6                    2018   
8848                      21                       6                    2018   
1956                      31                       1                    2018   
1184                      10                       1                    2018   
745                        4                       1                    2018   
6618                      11                       5                    2018   
6473                      17                       5                    2018   
6486                      16                       5                    2018   

      monto_prestado  
6882        200000.0  
9097        200000.0  
8580        200000.0  
8848        300000.0  
1956        200000.0  
1184        200000.0  
745         200000.0  
6618        200000.0  
6473        200000.0  
6486        300000.0

In [ ]:
muestreo15=df.sample(n=15)

**Toma 15 muestras aleatorias de todo la  población, en este caso 12152** 

In [ ]:
muestreo15

edad       sexo        cabeza_de_hogar caracterizacion_ciudadano  \
1104     22   femenino                     no                      otro   
5844     45   femenino                     no             independiente   
517      52   femenino                     no             independiente   
10560    29  masculino                     no                       NaN   
11598    80  masculino  aporta economicamente                       NaN   
1406     48   femenino                     si             independiente   
1862     42   femenino                     no             independiente   
11557    70   femenino  aporta economicamente                       NaN   
5407     28  masculino                     si               emprendedor   
10426    71   femenino                     no                       NaN   
4188     19  masculino                     no               emprendedor   
2540     24   femenino                     no                      otro   
10869    82   femenino  aporta economicamente                       NaN   
10419    58   femenino                     no                       NaN   
8338     23   femenino                     no                      otro   

       estrato            barrio_vereda                   comuna  \
1104       2.0        villa del socorro            02-santa cruz   
5844       3.0              60-la palma         60-san cristobal   
517        3.0              san joaquin              11-laureles   
10560      NaN                  no dato                      NaN   
11598      2.0        comuna 07 robledo               07-robledo   
1406       2.0     san antonio de prado  80-san antonio de prado   
1862       2.0       santa rosa de lima            13-san javier   
11557      3.0  comuna 08 villa hermosa         08-villa hermosa   
5407       2.0                 castilla              05-castilla   
10426      NaN                  no dato                      NaN   
4188       3.0                 castilla              05-castilla   
2540       2.0        villa del socorro            02-santa cruz   
10869      3.0   comuna 09 buenos aires          09-buenos aires   
10419      NaN                  no dato                      NaN   
8338       2.0          san javier no.1                 16-belén   

                    zona  fecha_de_beneficio_dia  fecha_de_beneficio_mes  \
1104         nororiental                      18                       1   
5844       corregimiento                       8                       5   
517    centro occidental                      12                       1   
10560                NaN                      23                      11   
11598      noroccidental                      15                       5   
1406       corregimiento                      30                       1   
1862   centro occidental                      25                       1   
11557    centro oriental                      13                       5   
5407       noroccidental                       4                       5   
10426                NaN                      15                       9   
4188       noroccidental                       3                       4   
2540         nororiental                       9                       2   
10869    centro oriental                      29                       4   
10419                NaN                      13                       9   
8338       suroccidental                      15                       6   

       fecha_de_beneficio_año  monto_prestado  
1104                     2018        300000.0  
5844                     2018        200000.0  
517                      2018        200000.0  
10560                    2017        200012.0  
11598                    2019        300000.0  
1406                     2018        200000.0  
1862                     2018        200000.0  
11557                    2019        300000.0  
5407                     2018        200000.0  
10426               

**Selecciona  de forma aleatoria el 10% de la población.** 

In [ ]:
# Selección aleatoria del 10% de la tabla
pob_10 = df.sample(frac =.1)
pob_10.head()

edad      sexo        cabeza_de_hogar caracterizacion_ciudadano  \
4172     39  femenino                     no             independiente   
11471    70  femenino  aporta economicamente                       NaN   
10863    70  femenino  aporta economicamente                       NaN   
687      39  femenino                     no               emprendedor   
1383     26  femenino                     no               emprendedor   

       estrato       barrio_vereda            comuna               zona  \
4172       2.0        buenos aires  10-la candelaria    centro oriental   
11471      3.0           el salado     13-san javier  centro occidental   
10863      2.0  comuna 04 aranjuez       04-aranjuez        nororiental   
687        2.0   manrique oriental       03-manrique        nororiental   
1383       3.0   campo valdes no.1       04-aranjuez        nororiental   

       fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
4172                        3                       4                    2018   
11471                      13                       5                    2019   
10863                      29                       4                    2019   
687                         5                       1                    2018   
1383                       16                       1                    2018   

       monto_prestado  
4172         300000.0  
11471        300000.0  
10863        300000.0  
687          200000.0  
1383         200000.0

In [ ]:
pob_10

edad       sexo        cabeza_de_hogar caracterizacion_ciudadano  \
4172     39   femenino                     no             independiente   
11471    70   femenino  aporta economicamente                       NaN   
10863    70   femenino  aporta economicamente                       NaN   
687      39   femenino                     no               emprendedor   
1383     26   femenino                     no               emprendedor   
...     ...        ...                    ...                       ...   
5596     59  masculino                     si             independiente   
2720     27   femenino                     no                      otro   
3471     34  masculino                     si                      otro   
1128     33   femenino                     si               emprendedor   
11197    84   femenino  aporta economicamente                       NaN   

       estrato          barrio_vereda              comuna               zona  \
4172       2.0           buenos aires    10-la candelaria    centro oriental   
11471      3.0              el salado       13-san javier  centro occidental   
10863      2.0     comuna 04 aranjuez         04-aranjuez        nororiental   
687        2.0      manrique oriental         03-manrique        nororiental   
1383       3.0      campo valdes no.1         04-aranjuez        nororiental   
...        ...                    ...                 ...                ...   
5596       3.0               pedregal  06-doce de octubre      noroccidental   
2720       2.0      manrique oriental         03-manrique        nororiental   
3471       2.0  manrique central no.1    10-la candelaria    centro oriental   
1128       1.0                robledo          07-robledo      noroccidental   
11197      2.0               medellin         05-castilla      noroccidental   

       fecha_de_beneficio_dia  fecha_de_beneficio_mes  fecha_de_beneficio_año  \
4172                        3                       4                    2018   
11471                      13                       5                    2019   
10863                      29                       4                    2019   
687                         5                       1                    2018   
1383                       16                       1                    2018   
...                       ...                     ...                     ...   
5596                        8                       5                    2018   
2720                       15                       2                    2018   
3471                        7                       3                    2018   
1128                       23                       1                    2018   
11197                      19                       3                    2019   

       monto_prestado  
4172         300000.0  
11471        300000.0  
10863        300000.0  
687          200000.0  
1383         200000.0  
...               ...  
5596         200000.0  
2720         200000.0  
3471         300000.0  
1128         200000.0  
11197        300000.0  

[1215 rows x 12 columns]

****El 10% de 12150 es 1215****

**Dimensión del data frame.**

In [ ]:
pob_10.shape

(1215, 12)

**Páginas para descargar datos libres.**

**Páginas para descargar datos libres Colombia.**

https://www.datos.gov.co/<br> https://datosabiertos.bogota.gov.co/<br> http://medata.gov.co/<br>
https://datosabiertos.metropol.gov.co/<br> https://datos.cali.gov.co/

**Páginas para descargar datos libres fuera de Colombia.**

https://www.kaggle.com/datasets <br>https://archive.ics.uci.edu/ml/index.php <br>https://www.data.gov/<br>
https://aws.amazon.com/datasets